### ISSUE TITLE
feat: Semantic Version comparison cop

### ISSUE THREAD

**Issue by Greg-Myers-SB** [rubocop/rubocop#14768]
## Is your feature request related to a problem? Please describe.

It's a common mistake for some engineers to do either string or float comparisons when comparing version numbers. I propose here a set of good and bad so that if there's interest we might be able to make a cop around this idea.
If one already exists somewhere then we can shine more light on it. :)

```ruby
# bad
version.to_f >= 2.2         # version = "2.10" fails due to truncation
version.to_i > 2            # version = 2.1 fails due to truncation
version >= "2.2"            # version = "2.10" fails due to string sort order
"2.2" <= version            # another syntax for the above
hash['version'].to_f >= 2.2 # another syntax for the above

# good
Gem::Version.new(version) >= Gem::Version.new(2.2)
```

## Describe the solution you'd like

A method of detecting string, int, and float comparisons that isn't extremely prone to false positives. Maybe it's hooked on comparison to magic string, or a variable name like "version" or "identifier" or "semver"

## Describe alternatives you've considered

-

## Additional context

-

**Comment by bbatsov** [comment]
Closing — semantic version comparison is too niche for core RuboCop. This could work as a custom cop or third-party extension.
